From 929821cbc22927a81ae1161c8467026a1e9d2a53 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Hundeb=C3=B8ll?= <mhu@silicom.dk>
Date: Mon, 1 Feb 2021 13:20:46 +0100
Subject: [PATCH] fpga: m10bmc-sec: add n5010 device id
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The Silicom n5010 PAC features the max10 bmc secure funtionality, so add
it to the list of device id's, and enable it in the intel-m10-bmc
multi-function driver.

Co-developed-by: Martin Hundebøll <mhu@silicom.dk>
Signed-off-by: Martin Hundebøll <mhu@silicom.dk>
Signed-off-by: Ilpo Järvinen <ilpo.jarvinen@linux.intel.com>
---
 drivers/fpga/intel-m10-bmc-sec-update.c | 4 ++++
 drivers/mfd/intel-m10-bmc-spi.c         | 1 +
 2 files changed, 5 insertions(+)

diff --git a/drivers/fpga/intel-m10-bmc-sec-update.c b/drivers/fpga/intel-m10-bmc-sec-update.c
index 33f1975bd0e7..c3a521b7fdfd 100644
--- a/drivers/fpga/intel-m10-bmc-sec-update.c
+++ b/drivers/fpga/intel-m10-bmc-sec-update.c
@@ -1123,6 +1123,10 @@ static const struct platform_device_id intel_m10bmc_sec_ids[] = {
 		.name = "d5005bmc-sec-update",
 		.driver_data = (kernel_ulong_t)&m10sec_d5005_ops,
 	},
+	{
+		.name = "n5010bmc-sec-update",
+		.driver_data = (kernel_ulong_t)&m10sec_d5005_ops,
+	},
 	{
 		.name = "n6000bmc-sec-update",
 		.driver_data = (kernel_ulong_t)&m10sec_n6000_ops,
diff --git a/drivers/mfd/intel-m10-bmc-spi.c b/drivers/mfd/intel-m10-bmc-spi.c
index 56e53847ccb3..6c2e17409783 100644
--- a/drivers/mfd/intel-m10-bmc-spi.c
+++ b/drivers/mfd/intel-m10-bmc-spi.c
@@ -132,6 +132,7 @@ static const struct regmap_range m10bmc_n3000_fw_handshake_regs[] = {
 
 static struct mfd_cell m10bmc_n5010_subdevs[] = {
 	{ .name = "n5010bmc-hwmon" },
+	{ .name = "n5010bmc-sec-update" },
 };
 
 static const struct intel_m10bmc_platform_info m10bmc_spi_n3000 = {
